frontend/src/studio-ui/studio-ui.tsx

```diff
@@ -0,0 +1,91 @@
+import * as React from 'react';
+import StylingPage from './styling-page';
+import EditingPage from './editing-page';
+
+enum ConfigPage {
+  Styling, Editing,
+}
+
+interface StudioUiProps {
+  initialPanels: Panel[]
+  initialStyling: PanelStyling
+  studioSaveUrl: string
+  runtime: XBlockRuntime
+}
+
+interface ActionButtonLinkProps {
+  className: string
+  onClick: ()=>void
+  label:string
+}
+function ActionButtonLink({ className, onClick, label }:ActionButtonLinkProps) {
+  return (
+    <li className="action-item">
+      {/* eslint-disable-next-line jsx-a11y/anchor-is-valid */}
+      <a
+        href="#"
+        className={`button action-primary ${className}`}
+        onClick={onClick}
+      >
+        {label}
+      </a>
+    </li>
+  );
+}
+
+export default function StudioUi({
+  initialPanels, initialStyling, studioSaveUrl, runtime,
+}: StudioUiProps) {
+  const [step, setStep] = React.useState(ConfigPage.Styling);
+  const [styling, setStyling] = React.useState(initialStyling);
+  const [panels, setPanels] = React.useState(initialPanels);
+  const handleSave = () => {
+    runtime.notify('save', { state: 'start' });
+    $.post(studioSaveUrl, JSON.stringify({ styling, panels }));
```

Comment:
It should be possible to use the fetch API however it will need more code and config. This is just readily available for use.

Reply:
I don't really see an issue. The platform promises jquery to XBlock so it's not going to go away in the short term. There is no problem to solve ehre.

Reply:
I agree that `$.post` is perfectly fine to use, but one disadvantage is that it's not using (real ES6) promises. If you want, I have a `fetch` based version here: https://github.com/open-craft/xblock-react/blob/fe1ae04d929983f02c052493e9c356fd8b1756bb/react_xblock_2/frontend/src/xblock-utils.ts#L47-L74